sql/catalyst/src/main/java/org/apache/spark/sql/connector/SupportsReportMetrics.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector;
+
+import org.apache.spark.annotation.Evolving;
+
+/**
+ * Data sources can implement this interface to
+ * report supported custom metrics to Spark in read/write path.
+ *
+ * @since 3.2.0
+ */
+@Evolving
+public interface SupportsReportMetrics {
+
+    /**
+     * Returns an array of supported custom metrics with name and description.
+     * By default it returns empty array.
+     */
+    default CustomMetric[] supportedCustomMetrics() {
+        CustomMetric[] NO_METRICS = {};
```

Comment:
shall we define it as a private static field?

Reply:
+1 for @cloud-fan 's suggestion to declare static field `NO_METRICS`.

Reply:
Unfortunately, Java interface cannot define private or protected field.